src/App.tsx

```diff
@@ -16,35 +18,77 @@
 ];
 
 export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
-        </button>
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setReverse] = useState(false);
+  const [isReset, setReset] = useState(goodsFromServer);
 
-        <button type="button" className="button is-success is-light">
-          Sort by length
-        </button>
+  enum SortType {
+    ALPHABETICALLY = 'Sort alphabetically',
+    LENGTH = 'Sort by length',
+  }
 
-        <button type="button" className="button is-warning is-light">
-          Reverse
-        </button>
+  const visibleGoods = [...goodsFromServer].sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.ALPHABETICALLY:
+        return good1.localeCompare(good2);
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
 
-        <button type="button" className="button is-danger is-light">
-          Reset
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  const handleReset = () => {
+    setReset(isReset);
```

Comment:
The `handleReset` function is not resetting the `visibleGoods` to its original state. Instead of setting `isReset`, you should directly reset the `sortField` and `isReversed` states, and ensure the goods list is reset to its original order.